domain/model/state/state.go

```diff
@@ -324,6 +325,39 @@ func (s *State) GetModelType(ctx context.Context, uuid coremodel.UUID) (coremode
 	return modelType, nil
 }
 
+// GetControllerModel returns the model the controller is running in.
+func (s *State) GetControllerModel(ctx context.Context) (coremodel.Model, error) {
+	db, err := s.DB()
+	if err != nil {
+		return coremodel.Model{}, errors.Trace(err)
+	}
+
+	controllerModelUUID := dbModelUUID{}
+	stmt, err := s.Prepare(`
+SELECT &dbModelUUID.model_uuid 
+FROM   controller
+`, controllerModelUUID)
+	if err != nil {
+		return coremodel.Model{}, errors.Annotate(err, "preparing get controller model UUID statement")
+	}
+
+	var model coremodel.Model
+	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		err := tx.Query(ctx, stmt).Get(&controllerModelUUID)
+		if errors.Is(err, sqlair.ErrNoRows) {
+			return errors.Annotatef(modelerrors.NotFound, "getting controller model")
```

Comment:
All hell has broken loose if this happens. Can we add a comment as to why this might be bad? Then we can point people in the right direction in the future.

Reply:
I've changed it to not be a `modeerrors.NotFound` error, and made it instead just a `fmt.Errorf`. On reflection, having an error type there implies that it is catch-able and something can be done about it, this is not the case.

I've added a comment too.